bundle/edu.gemini.shared.gui/src/main/scala/edu/gemini/shared/gui/textComponent/NumberField.scala

```diff
@@ -27,7 +27,7 @@ class NumberField(d: Option[Double]) extends FormattedTextField(NumberField.df)
   private val pink = new Color(255, 224, 224)
   private val white = background // N.B. this means we can't change the background to anything else
 
-  var valid = true
+  @volatile var valid = true
```

Comment:
Sorry, that was part of my ongoing investigation of why the FormattedTextField emits an extra empty event that confuses clicking on the ok button (REL-2253). As you say shouldn't be necessary as long as we are in Swing